drivers/i2c/i2c_ll_stm32.c

```diff
@@ -53,9 +56,23 @@ int i2c_stm32_runtime_configure(const struct device *dev, uint32_t config)
 	data->dev_config = config;
 
 	k_sem_take(&data->bus_mutex, K_FOREVER);
+
+#if IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME)
+	(void)pm_device_runtime_get(dev);
+#else
+	pm_device_busy_set(dev);
+#endif
```

Comment:
Should be:
```
#ifdef CONFIG_PM_DEVICE_RUNTIME
	(void)pm_device_runtime_get(dev);
#else
	pm_device_busy_set(dev);
#endif
```

Though, I wonder if the #idef is actually needed.

Another option would be:
```
	(void)pm_device_runtime_get(dev);
	pm_device_busy_set(dev);
```
This should compile in all combinations and make the code smoother to read.
There might be an overhead but I don't think it is killing.

@ceolin, @gmarull do you have any advice on the preferred implementation if we aim to implement `PM_DEVICE_RUNTIME` support on such driver ?

Reply:
I thought about it but then wondered if we should call `pm_device_busy_set` with `CONFIG_PM_DEVICE_RUNTIME` so that's why I have either one or this other.

Reply:
@ceolin @gmarull any advice?

Reply:
@fouge Have you been able to verify that both configurations work as expected ?

Reply:
It runs but I am not able to measure any significant drop in current consumption.

I obtain a baseline of ~90uA overall when using `PM_DEVICE` on my system, and 1.25mA with `PM_DEVICE_RUNTIME`, which is likely due to some other factors.

Reply:
Did you try to prevent core going to stop/sleep states ?